crates/sui-adapter/src/programmable_transactions/execution.rs

```diff
@@ -81,14 +82,31 @@ pub fn execute<S: StorageView, Mode: ExecutionMode>(
     // execute commands
     let mut mode_results = Mode::empty_results();
     for (idx, command) in commands.into_iter().enumerate() {
-        execute_command::<_, Mode>(&mut context, &mut mode_results, command)
-            .map_err(|e| e.with_command_index(idx))?
+        if let Err(err) = execute_command::<_, Mode>(&mut context, &mut mode_results, command)
+                        .map_err(|e| e.with_command_index(idx)) {
+                let object_runtime: &ObjectRuntime = context.session.get_native_extensions().get();
+                // We still need to record the loaded child objects for debugging
+                let loaded_child_objects = object_runtime.loaded_child_objects();
+                drop(context);
+                state_view.save_loaded_child_objects(loaded_child_objects);
+                return Err(err);
+        };
     }
+
+    // Save loaded objects table in case we fail in post execution
+    let object_runtime: &ObjectRuntime = context.session.get_native_extensions().get();
+    // We still need to record the loaded child objects for debugging
+    let loaded_child_objects = object_runtime.loaded_child_objects();
+
     // apply changes
+    let finished  = context.finish::<Mode>();
+    // Save loaded objects for debug. We dont want to lose the info
+    state_view.save_loaded_child_objects(loaded_child_objects);
+
```

Comment:
I think this can be reverted now? Or is there a reason to save the objects after finish rather than before?

Reply:
Rust borrow rules. Cant borrow `state_view` as mutable while it is borrow as immutable by `context`